File: client/models/Item.ts
export interface Item {
  name: string;
  description: string;
  price: number;
  image: string;
}

export function instanceOfItem(value: object): boolean {
  let isInstance = true
  isInstance = isInstance && "name" in value
  isInstance = isInstance && "description" in value
  isInstance = isInstance && "price" in value
  isInstance = isInstance && "image" in value

  return isInstance
}

export function ItemFromJSON(json: any): Item {
  return ItemFromJSONTyped(json, false)
}

export function ItemFromJSONTyped(json: any, ignoreDiscriminator: boolean): Item {
  if ((json === undefined) || (json === null)) {
    return json
  }
  return {

    "name": json["name"],
    "description": json["description"],
    "price": json["price"],
    "image": json["image"]
  }
}

export function ItemToJSON(value?: Item | null): any {
  if (value === undefined) {
    return undefined
  }
  if (value === null) {
    return null
  }
  return {

    "name": value.name,
    "description": value.description,
    "price": value.price,
    "image": value.image
  }
}

File: server/pkg/model/item.go
package model

import "gorm.io/gorm"

type Item struct {
	gorm.Model
	Name        string  `json:"name"`
	Description string  `json:"description"`
	Price       float32 `json:"price"`
	Image       string  `json:"image"`
}

func GetItem(id int) (*Item, error) {
	var item Item
	result := db.Find(&item, id)
	if result.Error != nil {
		return nil, result.Error
	}
	return &item, nil
}

func GetAllItems() (*[]Item, error) {
	var items []Item
	result := db.Find(&items)
	if result.Error != nil {
		return nil, result.Error
	}
	return &items, nil
}

func CreateItem(item Item) (*Item, error) {
	result := db.Create(&item)
	if result.Error != nil {
		return nil, result.Error
	}
	return &item, nil
}

func UpdateItem(item Item) (*Item, error) {
	result := db.Save(item)
	if result.Error != nil {
		return nil, result.Error
	}
	return &item, nil
}

func DeleteItem(id int) error {
	result := db.Delete(&Item{}, id)
	if result.Error != nil {
		return result.Error
	}
	return nil
}
